luarules/gadgets/cmd_build_bugger_off.lua

```diff
@@ -0,0 +1,204 @@
+        for builderID, _ in pairs(watchedBuilders) do
+
+            local firstCommand = getFirstCommand(builderID)
+            local targetID     = Spring.GetUnitIsBuilding(builderID)
+            local isBuilding   = false
+            if targetID then
+                isBuilding = true
+            end
+
+            local x, y, z = Spring.GetUnitPosition(builderID)
+            if firstCommand == nil or firstCommand.id > -1 or isBuilding then
+                print("Clearing watched builder fast")
+                watchedBuilders[builderID]      = nil
+                builderRadiusOffsets[builderID] = nil
+            elseif distance(firstCommand.params, x, z) > FAST_UPDATE_RADIUS then
+                print("Too far, demoting to slow")
+                watchedBuilders[builderID]    = nil
+                slowUpdateBuilders[builderID] = true -- Do distance checks less frequently
+            elseif distance(firstCommand.params, x, z) > BUILDER_BUILD_RADIUS then
+                -- Check distance frequently once you're closer
+            else
+                local cmdID           = firstCommand.id
+                local cmdParams       = firstCommand.params
+                local builtUnitDefID  = cmdID * -1
+                local builderTeam     = Spring.GetUnitTeam(builderID)
+                local buggerOffRadius = cachedUnitDefs[builtUnitDefID].radius + builderRadiusOffsets[builderID]
+
+                -- Get list of units to check
+                local targetX, targetY, targetZ = cmdParams[1], cmdParams[2], cmdParams[3]
+                local searchRadius     = cachedUnitDefs[builtUnitDefID].radius + SEARCH_RADIUS_OFFSET
+                local x1, x2, z1, z2   = targetX - searchRadius, targetX + searchRadius, targetZ - searchRadius, targetZ + searchRadius
+                local interferingUnits = Spring.GetUnitsInRectangle(x1, z1, x2, z2)
+
+                -- Escalate the radius every update. We want to send units away the minimum distance, but  
+                -- if there are many units in the way, they may cause a traffic jam and need to clear more room.
+                builderRadiusOffsets[builderID] = builderRadiusOffsets[builderID] + 5 
+
+                for i = 1, #interferingUnits do
+                    local interferingUnitID = interferingUnits[i]
+                    if builderID ~= interferingUnitID then
+                        local unitPosition = {Spring.GetUnitPosition(interferingUnitID)}
+                        local unitTeam     = Spring.GetUnitTeam(interferingUnitID)
+        
+                        if shouldIssueBuggeroff(unitTeam, builderTeam, interferingUnitID, targetX, targetY, targetZ, buggerOffRadius) then
+                            local sendX, sendZ = closestPointOnCircle(targetX, targetZ, buggerOffRadius, unitPosition[1], unitPosition[3])
+                            Spring.GiveOrderToUnit(interferingUnitID, CMD.INSERT, {0, CMD.MOVE, CMD.OPT_INTERNAL, sendX, targetY, sendZ}, {"alt"} )
```

Comment:
Microoptimisation to avoid allocating a table:
```suggestion
                            Spring.GiveOrderToUnit(interferingUnitID, CMD.INSERT, {0, CMD.MOVE, CMD.OPT_INTERNAL, sendX, targetY, sendZ}, CMD.OPT_ALT )
```